tests/src/testcommon.hh

```diff
@@ -236,29 +233,41 @@ template <typename NonLinearOperator>
   return t;
 }
 
-template <Ikarus::ResultType resType>
-[[nodiscard]] auto checkCalculateAt(auto& nonLinearOperator, auto& fe, const auto& feRequirements,
+template <template <typename, int, int> class resType, bool vectorizedResult = true>
+[[nodiscard]] auto checkCalculateAt(auto& /*nonLinearOperator*/, auto& fe, const auto& feRequirements,
                                     const auto& expectedResult, const auto& evaluationPositions,
                                     const std::string& messageIfFailed = "") {
   Dune::TestSuite t("Test of the calulateAt function for " + Dune::className(fe));
 
+  using FiniteElement = std::remove_cvref_t<decltype(fe)>;
   Eigen::MatrixXd computedResults(expectedResult.rows(), expectedResult.cols());
-  for (int i = 0; const auto& pos : evaluationPositions) {
-    auto result              = fe.template calculateAt<resType>(feRequirements, pos);
-    computedResults.row(i++) = result.transpose();
-  }
 
-  const bool isResultCorrect = isApproxSame(computedResults, expectedResult, 1e-8);
-  t.check(isResultCorrect) << "Computed Result for " << toString(resType) << " is not the same as expected result:\n"
-                           << "It is:\n"
-                           << computedResults << "\nBut should be:\n"
-                           << expectedResult << "\n"
-                           << messageIfFailed;
+  if constexpr (FiniteElement::template canProvideResultType<resType>()) {
+    for (int i = 0; const auto& pos : evaluationPositions) {
+      if constexpr (vectorizedResult) {
+        auto result              = fe.template calculateAt<resType>(feRequirements, pos).asVec();
+        computedResults.row(i++) = result.transpose();
+      } else {
+        auto result = fe.template calculateAt<resType>(feRequirements, pos).asMat();
+        Eigen::Map<Eigen::VectorXd> vector(result.data(), result.size());
```

Comment:
Why this inderiction couldn't you do `computedResults.row(i++) = result.reshaped().transpose();`